lib/stack_master/commands/tidy.rb

```diff
@@ -0,0 +1,68 @@
+module StackMaster
+  module Commands
+    class Tidy
+      include Command
+      include StackMaster::Commands::TerminalHelper
+
+      def initialize(config)
+        @config = config
+      end
+
+      def perform
+        used_templates = []
+        used_parameter_files = []
+
+        templates = Set.new(find_templates())
+        parameter_files = Set.new(find_parameter_files())
+
+        status = @config.stacks.each do |stack_definition|
+          parameter_files.subtract(stack_definition.parameter_files)
+          template = File.absolute_path(stack_definition.template_file_path)
+
+          if template
+            templates.delete(template)
+
+            if !File.exist?(template)
+              StackMaster.stdout.puts "Stack \"#{stack_definition.stack_name}\" in \"#{stack_definition.region}\" missing template \"#{rel_path(template)}\""
+            end
+          end
+        end
+
+        templates.each do |path|
+          StackMaster.stdout.puts "#{rel_path(path)}: no stack found for this template"
+        end
+
+        parameter_files.each do |path|
+          StackMaster.stdout.puts "#{rel_path(path)}: no stack found for this parameter file"
+        end
+      end
```

Comment:
Can we return a non-zero exit code if any of them is not empty? That would be nice for CI.

Reply:
Thought about it, but they're mostly indications rather than errors - eg in the data platform repositories, there's multiple stack_master.yml files that share a template directory.

They'll both find "extra" files as some templates are only used by one of the two configs .. but they're exceptions rather than the rule.

Reply:
I wonder if we should have a flag to enable or disable checking for orphaned templates?

Reply:
Maybe - my intention for this wasn't as a CI pass/fail tool but a convenience one, especially as the method for knowing which directories/files are referenced is not 100% reliable.